File: 008Verify_palindrome/008Verify_palindrome.py

class Solution:
    def isPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        while i < j:
            while i < j and not s[i].isalnum():
                i += 1
            while i < j and not s[j].isalnum():
                j -= 1
            if s[i].lower() != s[j].lower():
                return False
            i += 1
            j -= 1
        return True



str_1 = "A man, a plan, a canal: Panama"
str_2 = "race a car"
lss = Solution()
print("{}:{}".format(str_1, lss.isPalindrome(str_1)))
print("{}:{}".format(str_2, lss.isPalindrome(str_2)))

File: 008Verify_palindrome/008Verify_palindrome.cpp
#include<string>
#include<iostream>
#include<algorithm>

using namespace std;

class Solution {
public:
    bool isPalindrome(string & s) {
        int i = 0;
        int j = s.size() - 1;
        while (i < j)
        {
            while (i < j && !isalnum(s[i]))
                ++i;
            while (i < j && !isalnum(s[j]))
                --j;
            if (toupper(s[i++]) != toupper(s[j--]))
                return false;
        }
        return true;               
    }
};

int main() {
    string str_1 = "A man, a plan, a canal: Panama";
    string str_2 = "race a car";
    Solution lss;
    cout << str_1 << ":" << lss.isPalindrome(str_1) << endl;
    cout << str_2 << ":" << lss.isPalindrome(str_2) << endl;
}